srtcore/api.cpp

```diff
@@ -2338,99 +2351,48 @@ int CUDT::getsockopt(
 
 int CUDT::setsockopt(SRTSOCKET u, int, SRT_SOCKOPT optname, const void* optval, int optlen)
 {
-   try
-   {
-      CUDT* udt = s_UDTUnited.locateSocket(u, s_UDTUnited.ERH_THROW)->m_pUDT;
-      udt->setOpt(optname, optval, optlen);
-      return 0;
-   }
-   catch (const CUDTException& e)
+   if (!optval)
    {
-      s_UDTUnited.setError(new CUDTException(e));
-      return ERROR;
+       return setError(MJ_NOTSUP, MN_INVAL, 0);
    }
-   catch (const std::exception& ee)
-   {
-      LOGC(mglog.Fatal, log << "setsockopt: UNEXPECTED EXCEPTION: "
-         << typeid(ee).name() << ": " << ee.what());
-      s_UDTUnited.setError(new CUDTException(MJ_UNKNOWN, MN_NONE, 0));
-      return ERROR;
-   }
-}
 
-int CUDT::send(SRTSOCKET u, const char* buf, int len, int)
-{
    try
    {
-      CUDT* udt = s_UDTUnited.locateSocket(u, s_UDTUnited.ERH_THROW)->m_pUDT;
-      return udt->send(buf, len);
+       CUDT* udt = s_UDTUnited.locateSocket(u, s_UDTUnited.ERH_THROW)->m_pUDT;
+       udt->setOpt(optname, optval, optlen);
+       return 0;
    }
    catch (const CUDTException& e)
    {
-      s_UDTUnited.setError(new CUDTException(e));
-      return ERROR;
-   }
-   catch (bad_alloc&)
-   {
-      s_UDTUnited.setError(new CUDTException(MJ_SYSTEMRES, MN_MEMORY, 0));
-      return ERROR;
+       s_UDTUnited.setError(new CUDTException(e));
+       return ERROR;
```

Comment:
setError() replacement could be used here as well

Reply:
Would be actually nice to check everywhere - I think the `[[nodiscard]]` attribute can help, so I'll add it.